test/test.js

```diff
@@ -443,6 +443,16 @@ suite('Body mixin', function() {
   })
 
   suite('json', function() {
+    if (mocha.runner) {
+      setup(function() {
+        runner.removeListener('uncaughtException', runner.uncaught)
```

Comment:
Yikes, this means that during this suite, uncaught exceptions won't be handled my mocha as test failure right? Is there a way around this? Does the Firefox global error handler really capture exceptions in Promises even though they might be handled by its `catch` method?